tests/pd/test_rpc_client.rs

```diff
@@ -264,94 +228,38 @@ fn test_restart_leader() {
         .unwrap()
         .unwrap();
 
-    // Get the random binded addrs.
-    let eps = server.bind_addrs();
-
-    // Kill servers.
-    drop(server);
-    // Restart them again.
-    let mgr = Arc::new(SecurityManager::new(&SecurityConfig::default()).unwrap());
-    let _server = MockServer::run_with_eps::<Service>(&mgr, eps, Arc::clone(&se), None);
+    // Stop servers and restart them again.
+    server.stop();
+    server.start(&mgr, eps);
 
     // RECONNECT_INTERVAL_SEC is 1s.
     thread::sleep(Duration::from_secs(1));
 
-    let region = client.get_region_by_id(region.get_id());
-    region.wait().unwrap();
+    let region = client.get_region_by_id(region.get_id()).wait().unwrap();
+    assert_eq!(region.unwrap().get_id(), region_id);
```

Comment:
It seems strange to need to double unwrap the `region`.

Reply:
Yes, it looks strange, but it's defined in PdClient :frowning_face:. I am plan to refactor the pd mod later this quarter.

https://github.com/pingcap/tikv/blob/7ed4f6a91f92cad5cd5323aaebe7d9f04b77cc79/src/pd/mod.rs#L138

Reply:
Okay. :) Thanks for this explanation.